webapp/src/components/intersections/ReportsPage.tsx

```diff
@@ -129,6 +129,28 @@ const Page = () => {
   // Usually query is done on backend with indexing solutions
   const paginatedLogs = applyPagination(logs, page, rowsPerPage)
 
+  // Inside the parent component
+  const [selectedReport, setSelectedReport] = useState<ReportMetadata | null>(null)
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const handleViewReport = (report: ReportMetadata) => {
+    setSelectedReport(report)
+    setIsModalOpen(true)
+  }
+
+  const handleCloseReportModal = () => {
+    setIsModalOpen(false)
+    setSelectedReport(null)
+  }
+
+  const handleReportGenerated = () => {
+    setOpenReportGenerationDialog(false)
+    const refreshTime = new Date(Date.now() + 15 * 60 * 1000)
+    toast.success(
+      `Reports usually take 10-15 minutes to generate. Please refresh the page at ${refreshTime.toLocaleTimeString()} to see the new report.`
```

Comment:
We spoke offline on this a little at least with the data generation portion, but is there any notification that the report is ready for download? Or is it solely on the user to come back in 15 minutes to check?

Reply:
Right now, just on the user to check. The report generation is asynchronous